### ISSUE TITLE
XmlSqlBinaryReader is not working in .Net6

### ISSUE THREAD

**Issue by WaynePaiMS** [dotnet/runtime#74852]
### Description

XmlSqlBinaryReader is not working in .NET6 but working in .NET5. Below is callstack

```
Microsoft.AnalysisServices.ResponseFormatException: The server sent an unrecognizable response.
---> System.Xml.XmlException: '', hexadecimal value 0x06, is an invalid character.
at System.Xml.XmlSqlBinaryReader.CheckText(Boolean attr)
at System.Xml.XmlSqlBinaryReader.ScanOverValue(BinXmlToken token, Boolean attr, Boolean checkChars)
at System.Xml.XmlSqlBinaryReader.ScanOverAnyValue(BinXmlToken token, Boolean attr, Boolean checkChars)
at System.Xml.XmlSqlBinaryReader.ScanOverValue(BinXmlToken token, Boolean attr, Boolean checkChars)
at System.Xml.XmlSqlBinaryReader.ScanAttributes()
at System.Xml.XmlSqlBinaryReader.ImplReadElement()
at System.Xml.XmlSqlBinaryReader.ReadDoc()
at System.Xml.XmlSqlBinaryReader.Read()
```

I found there is a bug in `CheckText` of `XmlSqlBinaryReader`, which introduced by this commit, https://github.com/dotnet/runtime/pull/43379/files?diff=split&w=0.
The original `end` is `_pos`, which is next token position, and `CheckText` will just check to that position. Now, `_data.AsSpan` takes `_end`, which is end position of entire buffer(_data). Hence, `CheckText` will scan and validate for entire buffer and then encounter unexpected char.

### Reproduction Steps

Please ping chpai@microsoft.com for repro steps

### Expected behavior

XmlSqlBinaryReader should work in .NET6

### Actual behavior

XmlSqlBinaryReader is not working in .NET6

### Regression?

Yes, it's due to https://github.com/dotnet/runtime/pull/43379/files?diff=split&w=0

### Known Workarounds

No workaround

### Configuration

.NET6

**Comment by stephentoub** [comment]
cc: @GrabYourPitchforks

**Comment by GrabYourPitchforks** [comment]
Following up with the team offline for testing & validation.

**Comment by layomia** [comment]
We won't fix this in 7.0 at this point but we'll work on a fix in 8.0 and backport as appropriate.

**Comment by GrabYourPitchforks** [comment]
Agree with Layomi's reasoning. This issue already considers a backport to 6.0, so including a 7.0 backport isn't significantly more work. Better not to risk destabilizing things before the upcoming RC snap.

**Comment by jeffhandley** [comment]
Once this is fixed, we will indeed need to backport the fix to .NET 7 and .NET 6 to unblock customers upgrading from .NET 5 to .NET 6/7.

**Comment by manojsingh** [comment]
Do we have an ETA for this fix to be in .NET 6? This is a major blocker for a big customer from moving forward on a huge data analytics pipeline.

**Comment by jeffhandley** [comment]
Hi @manojsingh; there's no specific timeline yet. Stay tuned here though for when we have a resolution in .NET 8. Once we have that, we will reach out to folks affected and seek validation via nightly builds and/or .NET 8 previews. With that validation, we would be able to pursue a backport to .NET 7 and .NET 6 at the next monthly servicing releases.
